### ISSUE TITLE
Issues I can't figure out with Jykell

### ISSUE THREAD

**Issue by natalie-macpherson** [natalie-macpherson/portfolio#15]
1. For my portfolio info not all my pieces have a link. How do I set up the layout so this element shows up when needed and is hidden when not needed? I am guessing I need to do something involving an if statement which triggers visibility.
   https://github.com/natalie-macpherson/portfolio/blob/gh-pages/_layouts/portfolio.html#L29-L35
2. Issues with images and alts. I'm trying to set up the layout that the images and alts would changed based on the portfolio piece. There are several images for each piece. I have watched your video on page loops and images but I'm not sure if this will work for me.
   https://github.com/natalie-macpherson/portfolio/blob/gh-pages/_layouts/portfolio.html#L54-L56
3. For some reason the css is not being applied to most of the content on all my pages on the small screen. Do you know of any reason why this might be happening?
   I think that is is for now.

@thomasjbradley

**Comment by thomasjbradley** [comment]
**No 1:**

You're right—an if-statement. You'd wrap the optional HTML in the if-statement, like this:

``` html
{% if page.project-link %}
  <li class="line">
    ⋮
  </li>
{% endif %}
```

---

**No. 2:**

First we’d need to specify the image URLs in the portfolio piece file, like this:

``` yaml
---
images:
  - image-1.jpg
  - image-2.jpg
  - image-3.jpg
---
```

Then, in our layout, we can use a loop, like this:

``` html
{% for img in page.images %}
  <img src=="{{img}}" alt="">
{% endfor %}
```

---

**No. 3:**

I'm not exactly sure if this is the problem, but there's an unclosed media query here:
https://github.com/natalie-macpherson/portfolio/blob/gh-pages/_includes/css/main.css#L347

Also, though it's not causing any problems, the `@viewport` stuff only needs to be on the website once, so keep it inside `main.css` but remove it from all the other CSS files.

**Comment by natalie-macpherson** [comment]
The images are still not working.

https://github.com/natalie-macpherson/portfolio/blob/gh-pages/_layouts/portfolio.html#L62-L64 
is what I have written for my layout.

https://github.com/natalie-macpherson/portfolio/blob/gh-pages/work/refresh.html#L14-L17 
is what I have written for the piece file

Also, people are telling me I need to have .yml files for each portfolio piece instead of an html file. If I do this the website breaks and the links do not work anymore. Do I have to include .yml files.

One more thing, I highlighted my navigation on my pages with the if statement but I want the "portfolio" to be highlighted when I am on individual portfolio piece. Is there a way to do this?

@thomasjbradley

**Comment by thomasjbradley** [comment]
In the YAML file, the image names need to be indented with two spaces, and have a dash, like this:

``` yml
images:
  - refresh-plans.jpg
  - refresh-plans.jpg
  - refresh-white.jpg
```

And, in the image tag, here:
https://github.com/natalie-macpherson/portfolio/blob/gh-pages/_layouts/portfolio.html#L63
Are the images in an `images` folder?

---

You don't necessarily have to have a single YAML file for every portfolio piece if you have individual pages. The multiple YAML files is for single page websites.

---

Nav highlighting, use the word `contains` for comparison instead of `==`:

``` html
{% if page.url contains 'work' %}current{% endif %}
```

**Comment by natalie-macpherson** [comment]
I am no longer getting errors in the terminal but the images are not showing up in my portfolio piece. 

No the images are not in an images folder but I just changed images to img. Also, if I wanted to add additional images would I just add the images name to the list in the YAML? I think in general images seem to be confusing to myself and when I have asked people in class other people seem a bit confused with it too.

https://github.com/natalie-macpherson/portfolio/blob/gh-pages/_layouts/portfolio.html

On a side note, the images are in different locations on the page if I am trying to do a loop how does the jekyll know which image to use? Does it know what image to use based on the order in which i place them in the YAML?

Everything else works. Thank you so much for your help on theses issues.

@thomasjbradley

**Comment by thomasjbradley** [comment]
There's a couple things I can see to fix with the images:
1. It should be `page.img` because that's how it's written at the top in the YAML, here:
   https://github.com/natalie-macpherson/portfolio/blob/gh-pages/work/refresh.html#L14
2. There's two equals signs after `src==`.
3. All the images are in a folder named `img`, which doesn't exist anywhere in the URL.
4. `{{site.baseurl}}` should also be in the URL.

So, this is the corrected version:

``` html
{% for img in page.img %}
  <img class="img-flex shadow" src="{{site.baseurl}}/img/{{img}}" alt="">
{% endfor %}
```

---

Different location of images: It would probably be best to make multiple loops, one for each section & then have multiple YAML entries, like `img-design`, `img-research`, etc.

``` yaml
img-research:
  - thing-1.jpg
  - thing-3.jpg
  - thing-3.jpg
img-design:
  - thing-1.jpg
  - thing-2.jpg
  - thing-3.jpg
```

---

Image order: Yes, the images will be spit out in the order they're written in the YAML file.

**Comment by natalie-macpherson** [comment]
Ok I think I figured it all out now. It was a lot of small little things that just messed everything up. It was a struggle.  Anyways thank you for your help Thomas, I appreciate it and sorry for all the questions. I really don't like it when I am unable to figure something out on my own.

@thomasjbradley

**Comment by natalie-macpherson** [comment]
Hi Thomas sorry for bothering you again.

I am having issues make the alt's work for my images. I was able to kind of figure it out but it is placing in the wrong information.

Things like this work but it is picking up the image file and not the information for the image 
https://github.com/natalie-macpherson/portfolio/blob/gh-pages/_layouts/portfolio.html#L66-L68

I tried to make an md file for one of the images but I am not sure if this is the right path to take in order to retrieve the information for each images
https://github.com/natalie-macpherson/portfolio/tree/gh-pages/portfolio

In your video about looping pages and images you had md files that I believe you got the information from but creating these md files and making sure the path to them is correct is unclear. In very concern that my files are becoming a mess and that I am not putting things in the correct places and doing things the correct way.

I have been asking others to help but MANY people in the class seem VERY confused and are not really able to help. Some people are afraid to ask for help due to us wanting to give you space for more important matters.

---

On a side note:
I think the biggest issue I am having overall, is understanding how the pathing works for this layout system. In your videos you kind refer to them but it is not exactly clear so a lot of the time when you are linking things together understand where things came from is unclear.

@thomasjbradley
